test/utils/utils.go

```diff
@@ -1,37 +1,204 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
 package utils
 
 import (
+	"bufio"
+	"bytes"
+	"context"
 	"fmt"
 	"os"
 	"os/exec"
+	"reflect"
 	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2" //nolint:golint,revive
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 
-	//nolint:golint,revive
-	. "github.com/onsi/ginkgo"
-	"gopkg.in/yaml.v2"
+	"github.com/opendatahub-io/odh-model-controller/internal/controller/utils"
 )
 
+const (
+	prometheusOperatorVersion = "v0.77.1"
```

Comment:
can't these be installed as part of the CI setup?

Reply:
I guess yes. But since we don't  have E2E CI, I didn't do any clean-up of this scaffolded code.

Reply:
okay.